src/app/screens/signBatchPsbtRequest/index.tsx

```diff
@@ -0,0 +1,763 @@
+  // const [isModalVisible, setIsModalVisible] = useState(false);
+  // const [currentStepIndex, setCurrentStepIndex] = useState(0);
+  // const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  // const [isConnectSuccess, setIsConnectSuccess] = useState(false);
+  // const [isConnectFailed, setIsConnectFailed] = useState(false);
+  // const [isTxApproved, setIsTxApproved] = useState(false);
+  // const [isTxRejected, setIsTxRejected] = useState(false);
+  const { search } = useLocation();
+  const params = new URLSearchParams(search);
+  const tabId = params.get('tabId') ?? '0';
+  // const request = decodeToken(requestToken) as any as SignTransactionOptions;
+  // const btcClient = useBtcClient();
+
+  const handlePsbtParsing = useCallback(
+    (item) => {
+      try {
+        return parsePsbt(selectedAccount!, item.inputsToSign, item.psbtBase64, network.type);
+      } catch (err) {
+        return '';
+      }
+    },
+    [selectedAccount, network.type],
+  );
+
+  const parsedPsbts = useMemo(
+    () => payload.psbts.map(handlePsbtParsing),
+    [handlePsbtParsing, payload.psbts],
+  );
+
+  const userReceivesOrdinalArr = useDetectOrdinalInSignBatchPsbt(parsedPsbts);
+
+  const checkIfMismatch = () => {
+    if (!parsedPsbts?.length) {
+      navigate('/tx-status', {
+        state: {
+          txid: '',
+          currency: 'BTC',
+          errorTitle: t('PSBT_CANT_PARSE_ERROR_TITLE'),
+          error: t('PSBT_CANT_PARSE_ERROR_DESCRIPTION'),
+          browserTx: true,
+        },
+      });
+    }
+    if (payload.network.type !== network.type) {
+      navigate('/tx-status', {
+        state: {
+          txid: '',
+          currency: 'BTC',
+          error: t('NETWORK_MISMATCH'),
+          browserTx: true,
+        },
+      });
+    }
+
+    const checkAddressMismatch = (input) => {
+      if (input.address !== btcAddress && input.address !== ordinalsAddress) {
+        navigate('/tx-status', {
+          state: {
+            txid: '',
+            currency: 'STX',
+            error: t('ADDRESS_MISMATCH'),
+            browserTx: true,
+          },
+        });
+      }
+    };
+
+    payload.psbts.forEach((psbt) => psbt.inputsToSign.forEach(checkAddressMismatch));
+  };
+
+  useEffect(() => {
+    checkIfMismatch();
+  }, []);
+
+  useEffect(() => {
+    if (parsedPsbts) {
+      let outputScriptDetected = false;
+
+      parsedPsbts.forEach((psbt) => {
+        if (!psbt) {
+          return;
+        }
+
+        if (psbt.outputs.some((output) => !!output.outputScript)) {
+          outputScriptDetected = true;
+        }
+      });
+
+      setHasOutputScript(outputScriptDetected);
+    }
+  }, [parsedPsbts]);
+
+  const onSignPsbtConfirmed = async () => {
+    // TODO: Remove the delay for production
+    const delay = (ms) => new Promise((res) => setTimeout(res, ms));
+
+    try {
+      if (isLedgerAccount(selectedAccount)) {
+        // setIsModalVisible(true);
+        return;
```

Comment:
if we are not enabling this feature for ledger yet we should display an error message here cc: @yknl

Reply:
We actually display it on this code line: https://github.com/secretkeylabs/xverse-web-extension/blob/c1de9d8cc615ae9ea8d81152b42ed9e9edfc4cd2/src/app/screens/signBatchPsbtRequest/index.tsx#L413